zone/guild.cpp

```diff
@@ -121,28 +121,49 @@ void Client::SendGuildRanks()
 		{
 			while(i < permissions)
 			{
-				auto outapp = new EQApplicationPacket(OP_GuildUpdateURLAndChannel,
-								      sizeof(GuildUpdateRanks_Struct));
-				GuildUpdateRanks_Struct *guuacs = (GuildUpdateRanks_Struct*) outapp->pBuffer;
-				//guuacs->Unknown0008 = GuildID();
-				strncpy(guuacs->Unknown0012, GetCleanName(), 64);
-				guuacs->Action = 5;
-				guuacs->RankID = j;
-				guuacs->GuildID = GuildID();
-				guuacs->PermissionID = i;
-				guuacs->PermissionVal = 1;
-				guuacs->Unknown0089[0] = 0x2c;
-				guuacs->Unknown0089[1] = 0x01;
-				guuacs->Unknown0089[2] = 0x00;
-				FastQueuePacket(&outapp);
-				i++;
+					auto outapp = new EQApplicationPacket(OP_GuildUpdateURLAndChannel,
+						sizeof(GuildUpdateRanks_Struct));
+					GuildUpdateRanks_Struct* guuacs = (GuildUpdateRanks_Struct*)outapp->pBuffer;
+					//guuacs->Unknown0008 = GuildID();
+					strncpy(guuacs->Unknown0012, GetCleanName(), 64);
+					guuacs->Action = GuildUpdatePermissions;
+					guuacs->RankID = j;
+					guuacs->GuildID = GuildID();
+					guuacs->PermissionID = i;
+					//				guuacs->PermissionVal = 1;
+					guuacs->PermissionVal = guild_mgr.CheckPermission(GuildID(), j, (GuildAction)i);
+					guuacs->Unknown0089[0] = 0x2c;
+					guuacs->Unknown0089[1] = 0x01;
+					guuacs->Unknown0089[2] = 0x00;
+					FastQueuePacket(&outapp);
+					i++;
 			}
 			j++;
 			i = 1;
 		}
 	}
 }
 
+void Client::SendGuildRankNames() 
+{
+	if (IsInAGuild() && (ClientVersion() >= EQ::versions::ClientVersion::RoF))
+	{
+		auto guild = guild_mgr.GetGuildByGuildID(GuildID());
+		for (int i = 1; i <= 8; i++)
+		{
+			auto outapp = new EQApplicationPacket(OP_GuildUpdateURLAndChannel, sizeof(GuildUpdateUCPStruct));
+			GuildUpdateUCPStruct* gucp = (GuildUpdateUCPStruct*)outapp->pBuffer;
+
+			gucp->payload.rank_name.rank = i;
+			strcpy(gucp->payload.rank_name.rank_name, guild->rank_names[i].c_str());
+			gucp->action = 4;
```

Comment:
Constant, what is action 4 ?

Reply:
Updated this and also the 8 a few lines above.